src/ds/d500/d500-auto-calibration.cpp

```diff
@@ -225,7 +225,17 @@ namespace librealsense
 
     void d500_auto_calibrated::set_calibration_table(const std::vector<uint8_t>& calibration)
     {
-        _calib_engine->set_calibration_table(calibration, _curr_calibration);
+        if (_curr_calibration.size() != sizeof(ds::table_header) && // First time setting table, only header set by get_calibration_table
```

Comment:
Is `_curr_calibration` here is same as in `d500_debug_protocol_calibration_engine`? Notice that `d500_debug_protocol_calibration_engine::get_calibration_table` still uses the member

Reply:
get_valibration_table remained in the engine, because it uses the debug protocol interface.
It gets the current_calibration as param - this is the same as the one used in the set_calibration_table method:
 std::vector<uint8_t> d500_auto_calibrated::get_calibration_table() const
 {
     return _calib_engine->get_calibration_table(_curr_calibration);
 }